xapian-core/api/registry.cc

```diff
@@ -107,6 +111,49 @@ register_object(map<string, T*> & registry, const T & obj)
 }
 
 template<class T>
+static inline void
+register_object(map<string, T*> & registry1, map<string, T*> & registry2,
+		const T & obj)
+{
+    string name = obj.name();
+    if (rare(name.empty())) {
+	throw Xapian::InvalidOperationError("Unable to register object - name() method returned empty string");
+    }
+
+    string short_name = obj.name();
+    if (rare(short_name.empty())) {
+	throw Xapian::InvalidOperationError("Unable to register object - short_name() method returned empty string");
+    }
```

Comment:
I think it's OK for a weighting scheme not to provide a short name, especially as allowing that means existing user weighting scheme should continue to work (they just won't provide the ability to be created from a string description).

I think just gating everything which sets or uses `r2` with `if (!short_name.empty())` achieves that.